tds/src/main/java/thredds/server/catalogservice/CatalogViewContextParser.java

```diff
@@ -82,19 +90,48 @@ public Map<String, Object> getDatasetViewContext(Dataset ds, HttpServletRequest
     return model;
   }
 
-  private void addBaseContext(Map<String, Object> model) {
+  // returns protocol, authority, context path, and catalog access string
+  private String getBaseCatUrl(String catUrl, HttpServletRequest req) {
```

Comment:
could be static I think 
```suggestion
  private static String getBaseCatUrl(String catUrl, HttpServletRequest req) {
```